src/colvarproxy_io.cpp

```diff
@@ -288,17 +277,29 @@ int colvarproxy_io::close_output_stream(std::string const &output_name)
     return cvm::error("Error: trying to access an output file/channel "
                       "from the wrong thread.\n", COLVARS_BUG_ERROR);
   }
-  std::list<std::ostream *>::iterator osi  = output_files.begin();
-  std::list<std::string>::iterator    osni = output_stream_names.begin();
-  for ( ; osi != output_files.end(); osi++, osni++) {
-    if (*osni == output_name) {
-      ((std::ofstream *) (*osi))->close();
-      delete *osi;
-      output_files.erase(osi);
-      output_stream_names.erase(osni);
-      return COLVARS_OK;
-    }
+
+  if (output_streams_.count(output_name) > 0) {
+    (reinterpret_cast<std::ofstream *>(output_streams_[output_name]))->close();
+    delete output_streams_[output_name];
+    output_streams_.erase(output_name);
   }
-  return cvm::error("Error: trying to close an output file/channel "
-                    "that wasn't open.\n", COLVARS_BUG_ERROR);
+
+  return COLVARS_OK;
+}
+
+
+int colvarproxy_io::close_output_streams()
+{
+  if (! io_available()) {
+    return COLVARS_OK;
+  }
+
+  for (std::map<std::string, std::ostream *>::iterator osi = output_streams_.begin();
+       osi != output_streams_.end();
+       osi++) {
+    (reinterpret_cast<std::ofstream *>(osi->second))->close();
```

Comment:
How does this reinterpret_cast behave if the underlying object is not of type ofstream? Could that call an invalid function pointer and cause a crash?

Reply:
Since the map is private, the only chance that the wrong type of object can be stored there is if a derived proxy class overrides functions inconsistently. It's always possible, but it'd be a one-time error that would be caught when implementing that derived proxy class.

Reply:
Wouldn't it be safer to use dynamic_cast for these?

Reply:
Ok, fair point. I was initially avoiding it because it is not as flexible with multiple inheritance, and we can't control what other classes will be used from here. But in the base one we do know that the inheritance is simple. Changing it.